lib/diff.ml

```diff
@@ -128,8 +131,27 @@ let module_type_fallback ~loc ~typing_env ~name ~reference ~current =
       Some (Module { mname = name; mdiff = Modified Unsupported })
 
 let type_expr typing_env reference current =
-  if Ctype.does_match typing_env reference current then None
-  else Some (Modified { reference; current })
+  let rec same_type_var_names reference current =
```

Comment:
This is not required. I think the whole function can instead be rewritten as:
```ocaml
let type_expr ~typing_env ?(ref_params=[]) ?(cur_params=[]) ~reference ~current () =
  if Ctype.is_equal typing_env true (ref_params@[reference]) (cur_params@[current]) then
    None
  else
    Some (Modified {reference; current})
```

There might be some fine tuning to do here, e.g. there might be some extra steps like cloning the type_expr before calling `is_equal` but in the end it should all work with a single call to `Ctype.is_equal`, no manual Tvar matching to do!